Python/compile.c

```diff
@@ -970,6 +970,9 @@ _PyCompile_ResolveNameop(compiler *c, PyObject *mangled, int scope,
         break;
     }
     if (*optype != COMPILE_OP_FAST) {
+        /* XXX: Currently if this is used to insert a new name into u_freevars when
+           there are already entries in u_cellvars then the wrong index will be put
+           into u_freevars causing a hard error downstream. */
```

Comment:
Can we have an assertion that this is not being done, instead of as comment?

Reply:
This gets triggered if the situation arises:

https://github.com/tom-pytel/cpython/blob/285be46d378126297595ee785d4e1e1ef2d447e1/Python/assemble.c#L537

Reply:
In that case I don't think we need the comment here. Maybe a brief comment next to the assertion (where one would look if they made this mistake).

Reply:
Done.